src/main/java/org/jabref/logic/util/io/FileNameUniqueness.java

```diff
@@ -41,4 +45,90 @@ public static String getNonOverWritingFileName(Path targetDirectory, String file
+        }
+
+        String originalFileName = fileNameWithoutDuplicated + extensionSuffix;
+        if (fileName.equals(originalFileName)) {
+            return false;
+        }
+
+        Path originalFile = Path.of(directory.toString(), originalFileName);
+        // deal with a very special case, when duplication does not happen, but the file name is end with something like " (1)" as it originally is
+        if (!Files.exists(directory.resolve(originalFileName))) {
+            return false;
+        }
+
+        Path duplicateFile = Path.of(directory.toString(), fileName);
+        int counter = 1;
+        while (true) {
+            if (com.google.common.io.Files.equal(originalFile.toFile(), duplicateFile.toFile())) {
+                if (duplicateFile.toFile().delete()) {
+                    dialogService.notify(Localization.lang("Dupilcate file with '%0', successfully delete the file '%1'", originalFileName, fileName));
+                } else {
+                    dialogService.notify(Localization.lang("Dupilcate file with '%0', fail to delete the file '%1'", originalFileName, fileName));
+                }
+                return true;
+            }
+
+            originalFileName = fileNameWithoutDuplicated +
+                    " (" + counter + ")"
+                    + extensionSuffix;
+            counter++;
+
+            if (originalFileName.equals(fileName)) {
+                dialogService.notify(Localization.lang("Duplicate file name but different content, keep the new file '%0'", fileName));
+                return false;
+            }
+
+            originalFile = Path.of(directory.toString(), originalFileName);
+            if (!Files.exists(directory.resolve(originalFileName))) {
+                return false;
+            }
+        }
+    }
+
+    /**
+     * recover the file name to origin if it has duplicate mark such as " (1)"
+     * It will change the String whose format is "xxxxxx (number)" into "xxxxxx", while return the same String when it does not match the format
+     *
+     * @param fileName Suggested name for the file without extensionSuffix, if it has duplicate file name with other file, it will end with something like " (1)"
+     * @return Suggested name for the file without extensionSuffix and duplicate marks such as " (1)"
+     */
+    private static String eraseDuplicateMarks(String fileName) {
+        int dotPosition1 = fileName.lastIndexOf(')');
+        if (dotPosition1 != fileName.length() - 1 || dotPosition1 <= 2) {
+            return fileName;
+        } else {
+            if (!Character.isDigit(fileName.charAt(dotPosition1 - 1))) {
+                return fileName;
+            }
+        }
+        int dotPosition = fileName.lastIndexOf('(');
+        if ((dotPosition > 0) && (dotPosition < (fileName.length() - 1))) {
+            fileName = fileName.substring(0, dotPosition - 1);
+        }
+        return fileName;
```

Comment:
Please replace the whole method body and use regular expressions.

Hint: `(.*) \(\d+\)` and https://stackoverflow.com/q/988655/873282